docs/cmake-presets.md

```diff
@@ -0,0 +1,272 @@
+
+`CMakePresets.json` is a recommended alternative to kits and variants files. See [Enable `CMakePresets.json` in the CMake Tools extension](#enable-cmakepresetsjson-in-the-cmake-tools-extension) to enable or disable `CMakePresets.json` integration in the CMake Tools extension.
+
+## Supported CMake and `CMakePresets.json` versions
+
+The CMake Tools extension supports `CMakePresets.json` and `CMakeUserPresets.json` files version 2 or higher. You can update your file version by incrementing the version field in the root object. For an example and more information, see [`CMakePresets.json`](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#format).
+
+CMake version 3.20 or higher is required when invoking CMake with `CMakePresets.json` (version 2 or higher) from the command line. CMake Tools reads and evaluates `CMakePresets.json` and `CMakeUserPresets.json`, and does not invoke CMake directly with the `--preset` option. This means CMake version 3.20 or higher is not strictly required when building with `CMakePresets.json` inside Visual Studio Code. Use at least CMake version 3.14 or higher.
+
+## Enable `CMakePresets.json` in the CMake Tools extension
+
+A new setting, `cmake.useCMakePresets`, has been added to `settings.json`:
+
+|Setting  |Description  |Accepted values  | Default value |
+|---------|---------|---------|---------|
+|`cmake.useCMakePresets` |  Drive CMake configure, build, and test | `true`, `false`, `auto` | `auto` |
+
+`auto` : evaluates to `true` if there's a `CMakePresets.json` file in the `cmake.sourceDirectory` in the active folder. It evaluates to `false` if there isn't a `CMakePresets.json` file in the `cmake.sourceDirectory` in the active folder. Set `cmake.useCMakePresets` to `true` or `false` to explicitly enable or disable `CMakePresets.json` integration for all CMake projects.
+
+## Configure and build
+
+You can configure and build your CMake project with a series of commands. Open the command palette in Visual Studio Code with `Ctrl+Shift+P`:
+
+![Command pallette: Select Configure Preset](images/command-palette.png)
+
+**CMake: Select Configure Preset** lists the union of non-hidden Configure Presets defined in `CMakePresets.json` and `CMakeUserPresets.json`. Select a Configure Preset to make it the active Configure Preset. This is the `configurePreset` used when CMake is invoked to generate the project build system. The active Configure Preset is displayed in the status bar.
+
+CMake Tools uses the value of `hostOS` in the Microsoft Visual Studio Settings vendor map to hide Configure Presets that don't apply to your platform. See the entry for `hostOS` in the table under [Visual Studio Settings vendor map](cmake-presets-json-reference.md#visual-studio-settings-vendor-map) for more information.
+
+## CMake: configure
```

Comment:
Capitalize Configure

Reply:
done. As a matter of Microsoft style, we don't capitalize all the words in a heading or title. Usually just the first one. But I think this is referring to something that is captilized this way, yes?

Reply:
Yes, this is referring to the command which has sentence capitalization and is capitalized this way in the paragraph below. `CMake: Configure`

Reply:
I misspoke. Commands use title capitalization, not sentence capitalization.

Reply:
done